### PR TITLE
Add: webhooks documentation

### PR DESCRIPTION
**Pull request by teesloane**
- Adds documentation on new webhook functionality
- [Preview link](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/feat-webhooks-preview/2.0/webhooks/)

Notes / Todo
- [ ] Add documentation for parternships in separate docs, and then link to it?
- [ ] Eventually, remove note about open preview
- [x] Address technical comments/questions in PR review .

### CODE REVIEW

File: jekyll/_cci2/webhooks.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+1. Click on **Project Settings**.
+1. In the sidebar of your Project Settings, click on **Webhooks**.
+1. Click **Add Webhook**.
+1. Fill out the Webhook form (the table below describes the fields and their intent):
+
+| Field                  | Required? | Intent                                                               |
+|------------------------|-----------|----------------------------------------------------------------------|
+| Webhook name           | Y         | The name of your webhook                                             |
+| URL                    | Y         | The URL the webhook will make POST requests to.                      |
+| Certificate Validation | Y         |                                                                      |
```
Comment (teesloane) [inline jekyll/_cci2/webhooks.md:40]: What might be the best way to describe Intent for `certificate validation`?

**Review by radhigulati — COMMENTED** [review]
I think we should take out "reverse API" and just say "A webhook allows you to connect..."

File: jekyll/_cci2/webhooks.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+version:
+- Cloud
+---
+
+## Overview
+{: #overview}
+
+A webhook, sometimes referred to as a _reverse API_, allows you to connect a
+platform you manage (either an API you create yourself, or a third party
+service) to a stream of _future events_ (or, _real-time information_).
```
Comment (labsquared) [inline jekyll/_cci2/webhooks.md:15]: Remove (or, _real-time information_).

File: jekyll/_cci2/webhooks.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+{: class="table table-striped"}
+
+## Event Specifications
+{: #event-specifications}
+
+CircleCI currently offers webhooks for the following events:
+
+| Event type         | Description                                              | Potential statuses                                       | Included sub-entities                          |
+|--------------------|----------------------------------------------------------|----------------------------------------------------------|------------------------------------------------|
+| ping               | Test ping event when the webhook is initially configured |                                                          | project, organization                          |
```
Comment (labsquared) [inline jekyll/_cci2/webhooks.md:52]: Ping isn't a webhook event - it's just a test you can do once a specific webhook has been set up. 

What if we made "Test your Webhook" its own section below Event Specifications, and briefly describe how to use ping there?

File: jekyll/_cci2/webhooks.md (hunk 2 above)
Comment (aengelberg) [reply jekyll/_cci2/webhooks.md:15]: Should "future events" be italicized or only "events"? Only the latter seems to me like a distinct concept worth emphasizing.

File: jekyll/_cci2/webhooks.md (hunk 1 above)
Comment (aengelberg) [reply jekyll/_cci2/webhooks.md:40]: "Ensure the receiving host has a valid SSL certificate before sending an event. Only leave this unchecked for testing purposes."

File: jekyll/_cci2/webhooks.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+## Common sub-entities
+{: #common-sub-entities}
+
+The next sections describe the payloads of different events offered with
+CircleCI webhooks. The schema of these webhook events will share often share
+data with other webhooks - we refer to these as common maps of data as
+"sub-entities". For example, when you receive an event payload for the
+`job-completed` webhook, it will contains maps of data for your a *project,
+organization, workflow and pipeline*.
```
Comment (aengelberg) [inline jekyll/_cci2/webhooks.md:81]: This list should include "job" since the example is a job-related webhook.

File: jekyll/_cci2/webhooks.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+can also be triggered manually through the API.
+
+A pipeline's overall status doesn't change after it is created, so there are no
+exclusively-pipeline-related events.
+
+Re-running a workflow (either via "Re-run from failed" or "Re-run from
+beginning") creates a new, distinct workflow in the same pipeline.
+
+A pipeline may have parameters supplied during a manual invocation, that affect
+the behavior of its workflows and jobs.
```
Comment (aengelberg) [inline jekyll/_cci2/webhooks.md:154]: When I wrote the introductory paragraphs for these three sections (Job, Workflow, Pipeline), it was mostly for the benefit of our initial partners who needed to know how our entity model compares to other CI providers. That context may not be necessary for the typical reader of this page. What do you think about keeping these descriptions short and linking to other relevant docs within the site?

**Review by aengelberg — COMMENTED** [review]
Thanks for writing this up! As a high level suggestion I think including a sample full payload somewhere in here would be really helpful. Also, feel free to take any of my grammatical suggestions with a grain of salt.

File: jekyll/_cci2/webhooks.md (hunk 5 above)
Comment (labsquared) [reply jekyll/_cci2/webhooks.md:154]: I'm a fan of Alex's idea if you think it makes sense @teesloane.

File: jekyll/_cci2/webhooks.md (hunk 4 above)
Comment (aengelberg) [reply jekyll/_cci2/webhooks.md:81]: Also, just noticed a typo `your a`

File: jekyll/_cci2/webhooks.md (hunk 5 above)
Comment (teesloane) [reply jekyll/_cci2/webhooks.md:154]: Yeah, I can definitely trim this down - I think I would like to leave a brief description of pipelines (as we don't have a page dedicated to what a pipeline is exactly 😓 ) and then we'll update the page when there is one with a proper link.

File: jekyll/_cci2/webhooks.md (hunk 3 above)
Comment (teesloane) [reply jekyll/_cci2/webhooks.md]: Good point. I added an additional step about dispatching a test ping event under "Setting up a hook"

**Review by labsquared — APPROVED** [review]
This looks great @teesloane! :shipitparrot:

**Review by aengelberg — APPROVED** [review]
LGTM

**Pull request merged**